Transition to Android 15 (API level 35)

advanpix opened the pull request:
Various changes to support Android 15 (API level 35) and to update the project to a newer version of Kotlin and other components. Mostly build scripts are changed. Minor changes to source code - to fix incompatibilities with updated components, reduce warnings, etc. Resolves #185 and #187.

I tried to avoid any big jumps in versions. Only minimally required changes to support the Android 15. 
Now project uses gradle 8.2.0, Kotlin 1.8.10. Compiled fine with Android Studio Ladybug Feature Drop | 2024.2.2 Patch 2. 
Tested on Android 11 & 16.

iOS module has also been updated with higher component versions, incl. Java. Newest MobiVM 2.3.24 required to build wallet for iOS (it is successor to RoboVM). Wallet builds successfully using XCode 26.1.1 + Android Studio 2024.2.2. Patch 2 + MobiVM 2.3.24. Runs without issues on recent iOS versions. 
Tested on iPhone 12 mini (iOS 18.6.2).

Close #187 
Close #185 
Close #189

advanpix committed:
Resolves #185 and adds few improvements to the android/build script.

advanpix committed:
Fixed the incorrect reference to mosaik version in main build script

advanpix committed:
Fixed ergo-appkit dependency in common-jvm/buid script. Newer version, switch to Java 1.8

The ergo-appkit_2.11:5.0.3 requires few changes to code for compatibility.

advanpix committed:
Updates for compatibility with ergo-appkit_2.11:5.0.3 and minor changes to minimize compiler warnings

advanpix committed:
Regenerated gradle/verification-metadata.xml file to reflect new packages

advanpix committed:
Transition to newer version of Kotlin, gradle, AGP, compose and other components. All building scripts are refactored, cleaned up and updated.

advanpix committed:
Fixed incompatibility with missing zxing_button_ok in new version of zxing-android-embedded

MrStahlfelge commented on android/build.gradle:
```diff
@@ -116,12 +120,17 @@ dependencies {
+
+    // 1.3.1, Kotlin 1.8.10
+    // https://github.com/JetBrains/compose-multiplatform/releases/tag/v1.3.1
+    implementation "androidx.compose.compiler:compiler:1.4.2"
```
It is better to have these versions defined as a variable. Even better now is using a toml file

advanpix replied:
Makes sense, but we use these versions only here (not critical?)

MrStahlfelge commented on common-jvm/build.gradle:
```diff
@@ -4,33 +4,42 @@ plugins {
 java {
-    sourceCompatibility = JavaVersion.VERSION_1_7
-    targetCompatibility = JavaVersion.VERSION_1_7
+    sourceCompatibility = JavaVersion.VERSION_1_8
```
Changing this will probably break ios build

advanpix replied:
I am not sure if this breaks the iso build, but there is no another way if we want Android 15.

MrStahlfelge commented on common-jvm/build.gradle:
```diff
@@ -4,33 +4,42 @@ plugins {
     }
-    // kiama needs an own build for JRE7 compatibility
+    
+    // Custom Kiama - should work fine on Java 8
```
changing the comment but not changing kiama? :D

MrStahlfelge commented on common-jvm/build.gradle:
```diff
@@ -4,33 +4,42 @@ plugins {
     }
-    // stick to 20 for JRE7 compatibility
+    
+    // Guava 20 works with Java 8 (could upgrade later if needed)
```
Same: all the excludes and redefinitions are not needed on Java 8. But, probably we will loose ios.

advanpix replied:
Cleaned-up the common-jvm/build.gradle with your suggestions.

advanpix committed:
Cleaned up build.gradle in common-jvm gradle. Removed Java 1.7 specifics.

advanpix committed:
Fixed status bar overlap on Android 15

We use platform-specific theme flag to opt out, restoring the legacy layout behavior where the app window starts correctly below the status bar.  Resolves #187

advanpix committed:
Added aapt2 to trusted packages

advanpix committed:
Updated workflow to use JDK 17 instead of JDK 11

Android 15 requires at least AGP 8.2.0 which needs JDK 17

advanpix committed:
Updated workflow to enable manual dispatch for testing in forked repositories

advanpix committed:
Added junit-bom to trusted packages

advanpix committed:
Updated list of verified packages

advanpix committed:
Updated verification-metadata.xml

advanpix committed:
Updated list of verified packages with desktop-jvm-linux-x64

advanpix committed:
Fixed Retrofit call adapter failure in release build

The release build was failing to establish external network connections due to a ProGuard/R8 issue that stripped generic type information required by Retrofit's CallAdapterFactory.

Adds '-keepattributes Signature' and other Retrofit-specific ProGuard/R8 rules to prevent code minification from stripping generic type information, resolving the 'Unable to create call adapter' error during API calls.

advanpix committed:
Fixed Gson deserialization failure in release build

Resolves the final minification issue that prevented successful JSON deserialization in the signed release APK.

The issue was caused by ProGuard/R8 obfuscating the fields of network data models, resulting in Gson failing to map JSON keys to class fields.

Basically this disabled sending transactions from wallet
